packages/alfa-rules/src/sia-r56/rule.ts

```diff
@@ -0,0 +1,166 @@
+import { Diagnostic, Rule } from "@siteimprove/alfa-act";
+import { Node, Role } from "@siteimprove/alfa-aria";
+import { Array } from "@siteimprove/alfa-array";
+import { Element, Namespace } from "@siteimprove/alfa-dom";
+import { Iterable } from "@siteimprove/alfa-iterable";
+import { List } from "@siteimprove/alfa-list";
+import { Map } from "@siteimprove/alfa-map";
+import { Option } from "@siteimprove/alfa-option";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Err, Ok } from "@siteimprove/alfa-result";
+import { Page } from "@siteimprove/alfa-web";
+
+import { expectation } from "../common/expectation";
+import { Group } from "../common/group";
+import { normalize } from "../common/normalize";
+
+import { hasRole, isIgnored } from "../common/predicate";
+
+const { and, equals, not } = Predicate;
+const { hasNamespace } = Element;
+
+export default Rule.Atomic.of<Page, Group<Element>>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r56.html",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return document
+          .descendants({ flattened: true, nested: true })
+          .filter(Element.isElement)
+          .filter(
+            and(
+              hasNamespace(equals(Namespace.HTML)),
+              not(isIgnored(device)),
+              hasRole(device, (role) => role.is("landmark"))
+            )
+          )
+          .reduce((groups, landmark) => {
+            // since we already have filtered by having a landmark role, we can
+            // safely get the role.
+            const role = Node.from(landmark, device).role.get()!;
```

Comment:
```suggestion
            const role = Node.from(landmark, device).role.get();
```

Assuming that `Node#role` is of type `Option<Role>` then a non-null assertion isn't needed as `Option<Role>#get()` will return `Role`.